Ch 444/ch 457 added a check for empty values

NatalliaBulyha opened the pull request:
jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiNThiNDVhN2NhYWJlNDlkN2E2ZGEzNzU5MDdlYTJmOGYiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed the name of the custom exception: AuthenticationException to AuthException

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDZjZThmNjhiNmUyNDZhMWI1YzE0MDY5MmIxNDdiZjciLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed JwtAuthEntryPoint, redirected the 401 error to the exception handler.

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDZjZThmNjhiNmUyNDZhMWI1YzE0MDY5MmIxNDdiZjciLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed ErrorResponse, added additional fields

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDZjZThmNjhiNmUyNDZhMWI1YzE0MDY5MmIxNDdiZjciLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed the logic of building the ErrorResponse

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDZjZThmNjhiNmUyNDZhMWI1YzE0MDY5MmIxNDdiZjciLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed method to catch valid exceptions

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiMzRhMmQ3ZjZjZDQyNDlmM2E5ZGI2OWMxM2RkMWU1MDQiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 added a check on the services

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiNThiNDVhN2NhYWJlNDlkN2E2ZGEzNzU5MDdlYTJmOGYiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 added a check on the services

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiNThiNDVhN2NhYWJlNDlkN2E2ZGEzNzU5MDdlYTJmOGYiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed of check in ReviewService

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDYzYzMwYjc3NTAyNGE3NmJiNjJmODE1NTI0YzgwMDQiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed of check in part of ProductService

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDYzYzMwYjc3NTAyNGE3NmJiNjJmODE1NTI0YzgwMDQiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed of check in ProductServiceImpl and in Builder

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDYzYzMwYjc3NTAyNGE3NmJiNjJmODE1NTI0YzgwMDQiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 changed CozyHomeExceptionHandler

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDYzYzMwYjc3NTAyNGE3NmJiNjJmODE1NTI0YzgwMDQiLCJwIjoiaiJ9

Commit:
CH-444/CH-457 added a past version of the JwtAuthEntryPoint

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZDYzYzMwYjc3NTAyNGE3NmJiNjJmODE1NTI0YzgwMDQiLCJwIjoiaiJ9

NatalliaBulyha committed:
Merge branch 'main' into CH-444/CH-457

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/handler/CozyHomeExceptionHandler.java:
```diff
@@ -24,59 +27,75 @@
+    @Value("${reflectoring.trace}")
+    private boolean printStackTrace;
+
+    @ExceptionHandler({DataNotFoundException.class, IllegalArgumentException.class, NoSuchElementException.class})
```
Do we have any difference in use between DataNotFoundException, IllegalArgumentEx. and NoSuchElementEx.?

NatalliaBulyha replied:
we need to write all the errors in ExceptionHandler we can get

DariaBoo replied:
My point is that we should have a single strategy of using DataNotFoundException or IllegalArgumentException

Commit:
CH-444/CH-657 added AuthException

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiZGFmM2E5ZDA0Mjk1NGY2ZDk2MDFmODNmMjQ2NWE0NDciLCJwIjoiaiJ9

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/reviewservice/service/ReviewServiceImpl.java:
```diff
@@ -29,7 +30,8 @@ public class ReviewServiceImpl implements ReviewService{
-            throw new DataNotFoundException("The are no reviews.");
+            log.info("[ON getReviews]:: The are no reviews for any product.");
+            return new ArrayList<>();
         }
```
instead of this if statement we can add filter method to stream to check null values:
"return reviews.stream().filter(review -> review != null).map(review -> mapper.map(review, ReviewAdminResponse.class)).toList();" if the reviews List is empty will be returned empty list

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/reviewservice/service/ReviewServiceImpl.java:
```diff
@@ -48,6 +50,7 @@ public ReviewResponse addNewReview(ReviewRequest reviewRequest, String userId) {
         if (reviews.isEmpty()) {
+            log.info("[ON getReviewsForProduct]:: Review for product with sku code = {} isn't exist.", productSkuCode);
             return new ArrayList<>();
         }
```
we can add filter method to stream in the ReviewBuilder:
"public List<ReviewResponse> buildReviewsResponse(List<Review> reviews) {
        return reviews.stream().filter(review -> review!=null).map(this::buildReviewResponse).toList();
    }"

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/reviewservice/service/ReviewServiceImpl.java:
```diff
@@ -70,7 +74,8 @@ public void removeReviewById(ReviewToRemoveDto reviewRemoveDto) {
+            log.info("[ON getReviewsForProductAllInf]:: Review for product with sku code = {} isn't exist.", productSkuCode);
+            return new ArrayList<>();
         }
         return reviews.stream().map(review -> mapper.map(review, ReviewAdminResponse.class)).toList();
```
the same, we can simple add filter method to stream to avoid NullPointerException

NatalliaBulyha replied:
done

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/JWT/JwtAuthEntryPoint.java:
```diff
@@ -1,24 +1,29 @@
+    @Override
     public void commence(HttpServletRequest request, HttpServletResponse response,
-            AuthenticationException authException) throws IOException, ServletException {
+                         AuthenticationException authException) {
```
this method will not throws IOException, ServletException? or why they was deleted?

NatalliaBulyha replied:
yes, with the current implementation of the method they are not active

Commit:
CH-444/CH-457 code change by comments from git

jira [add exception handling where needed] https://cozyhometeam.atlassian.net/browse/CH-457?atlOrigin=eyJpIjoiYTQ5YTlmZjIyY2RkNDkxOWJjZmUwNTA1NzgyNWY3YTgiLCJwIjoiaiJ9

NatalliaBulyha committed:
Merge branch 'main' into CH-444/CH-457

Commit:
resolving conflicts
